soc/arm/st_stm32/stm32h7/Kconfig.soc

```diff
@@ -73,3 +73,7 @@ config SOC_STM32H7B3XXQ
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
 endchoice
+
+config STM32H7_WAKE_M4
```

Comment:
I agree with replacing `WAKE` by `BOOT` and also mentioning `M4` first is totally fine.

The reference to the application confuses me because there are two other options: M4 boot is triggered by hardware with `BCM4=1` or not at all what is at least possible and could be a power saving feature. Furthermore referencing the boot by application refers to what this option is _not doing_ instead of what it _is doing_.

Thus I would propose something like `STM32H7_M4_BOOT_BY_KERNEL` and "Boot M4 core by M7 kernel." with `default y`. I could also think of the order `STM32H7_BOOT_M4_BY_KERNEL` what is closer to natural English grammar but this is not that important.

What do you think?

Reply:
Oh this is good to know! I like the name you suggested. I will force-push the new version in a few minutes.

Reply:
The current version should do the job but the flag definition is still in `Kconfig.soc`. If I move it to `Kconfig.series` as you suggested it becomes an option for SOC selection.

Do you know how to avoid this? I am currently looking trough the Kconfig files and documentation but did not yet find a reason why this behaves like it does.

Reply:
> If I move it to `Kconfig.series` as you suggested it becomes an option for SOC selection.

Can you clarify ?

Reply:
I just pushed as a WIP commit what I meant. It does not seem to work. You could maybe point me to the right direction if you have a quick idea. Otherwise I will have a closer look probably next week.

Reply:
Sorry, I am a bit late. This is now solved and pushed.